stan/math/rev/scal/fun/beta.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef STAN_MATH_REV_SCAL_FUN_BETA_HPP
+#define STAN_MATH_REV_SCAL_FUN_BETA_HPP
+
+#include <stan/math/rev/core.hpp>
+#include <stan/math/prim/scal/fun/beta.hpp>
+#include <stan/math/prim/scal/fun/digamma.hpp>
+
+namespace stan {
+namespace math {
+
+/*
+ * Returns the beta function and its gradients.
+ *
+   \f[
+     \mathrm{beta}(a,b) = \left(B\left(a,b\right)\right)
+   \f]
+
+   \f[
+    \frac{\partial }{\partial a} = \left(\psi^{\left(0\right)}\left(a\right)
+                                      - \psi^{\left(0\right)}\left(a +
+ b\right)\right)
+                                    * \mathrm{beta}(a,b)
+   \f]
+
+   \f[
+    \frac{\partial }{\partial b} = \left(\psi^{\left(0\right)}\left(b\right)
+                                      - \psi^{\left(0\right)}\left(a +
+ b\right)\right)
+                                    * \mathrm{beta}(a,b)
+   \f]
+ *
+ * @tparam T1 Type of a argument
+ * @tparam T2 Type of b argument
+ * @param a Argument
+ * @param b Argument
+ * @return Result of beta function
+ */
+namespace internal {
+class beta_vv_vari : public op_vv_vari {
+ public:
+  beta_vv_vari(vari* avi, vari* bvi)
+      : op_vv_vari(beta(avi->val_, bvi->val_), avi, bvi) {}
+  void chain() {
+    const double beta_ab = beta(avi_->val_, bvi_->val_);
```

Comment:
Th value of `beta(avi_->val_, bvi_->val_)` is stored in the field `val_`, so it doesn't need to be recomputed.  So rather than using `beta_ab` here, just use `val_` (or maybe `this -> val_` as the compiler sometimes has trouble resolving member variables for reasons I don't understand.